### PR TITLE
core-9110 Allow users to copy multiple urls. Refactor code to use 

### PR DESCRIPTION
**Pull request by sri2k1us**
Clipboard copy feature.

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java [hunk 1]
```diff
@@ -25,27 +29,81 @@
...
+    private String footerText;
+    private String promptText;
 
-    public ClipboardCopyEnabledDialog() {
-        HorizontalPanel panel = new HorizontalPanel();
-        panel.setSpacing(5);
+    Appearance appearance = GWT.create(Appearance.class);
 
-        textBox = new TextBox();
+    public ClipboardCopyEnabledDialog(boolean copyMultiLine, boolean contextualHelpTool) {
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java]: Hmm I was thinking about. Do you know if it possible to do `@Assisted` parameters with `AsynProviderWrapper` ? I couldn't find a example of that in our code base.

[the remark this answers is not in the document]
Comment (ash3rz) [reply de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java]: If there is a way, I haven't seen it.  I think with most dialogs though, anything passed into the constructor can usually wait to be set in the `show()` method if that's where you're getting stuck.  The [CreateFolderDialog](https://github.com/cyverse-de/ui/blob/master/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/dialogs/CreateFolderDialog.java#L46) does it for example.

[the remark this answers is not in the document]
Comment (sri2k1us) [reply de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java]: `IplantDialog` base class requires `contextualHelpTool` boolean in constructor. Refactoring `IplantDialog` will requires changes to 100's of classes. I would save that for a separate PR

[the remark this answers is not in the document]
Comment (ash3rz) [reply de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java]: Sure, it can wait for another PR 👍

**Pull request merged**